packages/hooks_riverpod/lib/src/user_provider_listener.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'internals.dart';
+
+/// A hook that listens to a provider and and triggers a callback whenevr the value of provider is changed.
+///
+/// As a usage example, consider:
+///
+/// ```dart
+/// class Counter extends StateNotifier<int> {
+///   Counter(): super(0);
+///    void increment() => state++;
+/// }
+///
+/// final counterProvider = StateNotifierProvider((ref) => Counter());
+/// ```
+///
+/// We can then use [useProviderListener] to listen to `helloWorldProvider` like so:
+///
+/// ```dart
+/// class Example extends HookWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     final counter = userProvider(counter.state);
+///     useProviderListener(counterProvider, (context, value) {
+///       if (value % 2 == 0) print('even');
+///       else print('odd');
+///     });
+///    return Column(
+///       children: [
+///         Text('$count'),
+///         RaisedButton(
+///           onPressed: () => context.read(counterProvider).increment(),
+///          child: const Text('increment'),
+///         )
+///       ],
+///     );
+///   }
+/// }
+/// ```
+///
+/// See also:
+///
+/// - [Provider]/`select`, for filtering unwanted rebuilds.
+T useProviderListener<T>(
```

Comment:
It should return `void` instead and not cause the widget to rebuild when the value updates